types/module/module.go

```diff
@@ -319,7 +320,14 @@ func (m *Manager) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, genesisData
 		}
 		ctx.Logger().Debug("running initialization for module", "module", moduleName)
 
-		moduleValUpdates := m.Modules[moduleName].InitGenesis(ctx, cdc, genesisData[moduleName])
+		// TODO: Is this the best way to get the module's genesis type?
+		moduleGenesis := m.Modules[moduleName].DefaultGenesis()
+		if moduleGenesis != nil {
+			if err := cdc.UnmarshalJSON(genesisData[moduleName], moduleGenesis); err != nil {
+				panic(fmt.Sprintf("failed to parse %s genesis state: %s", moduleName, err))
+			}
+		}
+		moduleValUpdates := m.Modules[moduleName].InitGenesis(ctx, moduleGenesis)
```

Comment:
Does this look ok to you?

Reply:
If we're okay with `moduleGenesis` potentially being `nil`, then yes. But then you may as well move the `moduleValUpdates := m.Modules[moduleName].InitGenesis(ctx, moduleGenesis)` call into the `if` statement.